Installation and Code Generation Setup Docs (#2517)
apollographql/apollo-ios: docs/source/get-started.mdx

Follow the steps below to add Apollo iOS to your app:

## 1. Install the Apollo frameworks

You can install Apollo iOS into your project using any of the three major Cocoa ecosystem package managers: Swift Package Manager, CocoaPods, or Carthage.

<a name="spm-package-installation" />
<SPMPackageInstallationPanel />

<a name="spm-xcode-installation" />
<SPMXcodeInstallationPanel />

<a name="cocoapods-installation" />
<PodsInstallationPanel />

<a name="carthage-installation" />
<CarthageInstallationPanel />

## 2. Add a schema file to your target directory

For Apollo iOS to generate models for your GraphQL operations, you need a local copy of your GraphQL server's schema. To acquire this schema, see [Downloading a schema](./downloading-schema/).

## 3. [Optional] Install the Xcode add-ons for syntax highlighting

If you would like to add syntax highlighting for GraphQL query document files to Xcode, install the Xcode GraphQL add-ons.

Check out the [`xcode-graphql`](https://github.com/apollographql/xcode-graphql) repository and follow its [installation guide](https://github.com/apollographql/xcode-graphql#installation).

## 4. Create `.graphql` files for your GraphQL operations

> Apollo iOS generates code from GraphQL queries and mutations defined in files in your target. In order to use Apollo iOS, you'll need to define at least one operation GraphQL operation.

Reviewer: ```suggestion
Apollo iOS generates code from the GraphQL queries and mutations defined in your target's files. To use Apollo iOS, you'll need to define at least one operation GraphQL operation.
```

GraphQL operation and fragment definitions traditionally have the file extension `.graphql`. The generated models will have the file extension `.graphql.swift`.

## 5. Setup and run code generation

> Apollo iOS code generation uses your `.graphql` files to generate API code that helps you execute all forms of GraphQL operations, as well as parse and cache operation responses.

Reviewer: ```suggestion
Apollo iOS code generation uses your `.graphql` files to generate API code that helps you execute GraphQL operations and parse and cache operation responses.
```

Whenever you make changes to your GraphQL operation definitions, you'll need to run the code generation engine to re-generate your GraphQL models.

The easiest way to do this is with the Codegen CLI provided with Apollo iOS.

> For more advanced usage and configuration (including use with modularized projects), see [Code Generation](./code-generation).
>
> To use Apollo's code generation and schema downloader from within any Swift script or library, check out [Swift Scripting](./swift-scripting).

> <a name="spm-run-codegen" />

Reviewer: Something weird is happening with this first `ExpansionPanelListItem` in this callout — it's not wrapping correctly or allowing scrolling, so half of the content isn't accessible or visible.

Author: Good  catch. This is a bug for @trevorblades I think. Any chance we can get this fixed quickly Trevor?

If you see this image, the text only in the first list item is not wrapping correctly and text is getting cut off. Also the code blocks in the list item does not allow scrolling so you can't see the rest of the code block!

The second list item is working correctly though.

![image](https://user-images.githubusercontent.com/6243461/193084285-be145a8e-c078-4165-aeb4-ce867103b775.png)

<SPMSetupCodegenPanel />

<a name="pods-run-codegen" />
<PodsSetupCodegenPanel />

## 6. Create an `ApolloClient`

Before you can execute GraphQL operations in your app, you need to initialize an `ApolloClient` instance.

```swift
import Foundation
import Apollo

let apolloClient = ApolloClient(url: URL(string: "http://localhost:4000/graphql")!)
```

## 7. Fetch a query

> Your generated operation definitions can be fetched with your `ApolloClient` and the response will be returned as a type-safe generated data model.

Reviewer: ```suggestion
`ApolloClient` can fetch your generated operation definitions, and its returned response is a type-safe generated data model.
```
